python/cudf/cudf/core/dataframe.py

```diff
@@ -163,6 +163,7 @@ def __init__(self, data=None, index=None, columns=None):
                 self.add_column(col_name, series, forceindex=index is not None)
 
         self._add_empty_columns(columns, index)
+        self._allow_setattr_to_setitem = True
```

Comment:
```suggestion
        # this flag marks the completion of `__init__`, which changes the behavior
        # of `__setattr__` to enable the Pandas-like attr functionality `df.x = col`.
        self._allow_setattr_to_setitem = True
```

Reply:
IMO `# allows Pandas-like __setattr__ functionality: 'df.x = column', etc.` is more appropriate, since it's possible to use this flag after `__init__` to patch in new attributes without worrying about overlapping column names.